supervision/tracker/byte_tracker/core.py

```diff
@@ -249,12 +252,16 @@ def update_with_detections(self, detections: Detections) -> Detections:
             detections.confidence = np.array(
                 [t.score for t in tracks], dtype=np.float32
             )
+            detections.mask = np.array([t.mask for t in tracks], dtype=bool)
+
         else:
             detections.tracker_id = np.array([], dtype=int)
 
         return detections
 
-    def update_with_tensors(self, tensors: np.ndarray) -> List[STrack]:
+    def update_with_tensors(
+        self, tensors: np.ndarray, masks: np.ndarray = None
```

Comment:
Let's use Optional typing here. So: `masks: Optional[np.ndarray] = None`

Reply:
I've done it )